ipv6/address.go

```diff
@@ -0,0 +1,124 @@
+package ipv6
+
+import (
+	"fmt"
+	"net"
+)
+
+const (
+	// addressSize is the number of bits that an IPv6 address takes
+	addressSize int = 128
+)
+
+// Address represents an IPv6 address
+type Address struct {
+	ui uint128
+}
+
+// AddressFromUint64 returns the IPv6 address from its two 64 bit unsigned representation
+func AddressFromUint64(high, low uint64) Address {
+	return Address{uint128{high, low}}
+}
+
+// AddressFromUint16 returns the IPv6 address from its eight 16 bit unsigned representation
+func AddressFromUint16(a, b, c, d, e, f, g, h uint16) Address {
+	high := uint64(a)<<48 |
+		uint64(b)<<32 |
+		uint64(c)<<16 |
+		uint64(d)
+	low := uint64(e)<<48 |
+		uint64(f)<<32 |
+		uint64(g)<<16 |
+		uint64(h)
+	return Address{uint128{high, low}}
+}
+
+// AddressFromNetIP converts
+func AddressFromNetIP(ip net.IP) (Address, error) {
+	return fromSlice(ip)
+}
+
+// ParseAddress returns the ip represented by `addr` in colon notation. If
+// the address cannot be parsed, then error is non-nil and the address returned
+// must be ignored.
+func ParseAddress(address string) (Address, error) {
+	netIP := net.ParseIP(address)
+	if netIP == nil {
+		return Address{}, fmt.Errorf("failed to parse IPv6: %s", address)
+	}
+
+	netIPv4 := netIP.To4()
+	if netIPv4 != nil {
+		return Address{}, fmt.Errorf("address is not IPv6: %s", address)
+	}
+	return AddressFromNetIP(netIP)
+}
+
+// minAddress returns the address, a or b, which comes first in lexigraphical order
+func minAddress(a, b Address) Address {
+	if a.lessThan(b) {
+		return a
+	}
+	return b
+}
+
+// maxAddress returns the address, a or b, which comes last in lexigraphical order
+func maxAddress(a, b Address) Address {
+	if a.lessThan(b) {
+		return b
+	}
+	return a
+}
+
+// ToNetIP returns a net.IP representation of the address which always has 4 bytes
+func (me Address) ToNetIP() net.IP {
+	return me.toBytes()
+}
+
+// Equal reports whether this IPv6 address is the same as other
+func (me Address) Equal(other Address) bool {
```

Comment:
I decided to remove this method from the ipv4 side realizing that it is completely redundant with `==`. I don't know, I guess when I first wrote this, I didn't fully appreciate what `comparable` was in golang. :)